src/apps/zeromq/gpsd_client/gpsd_client.cpp

```diff
@@ -72,75 +74,65 @@ goby::apps::zeromq::GPSDClient::GPSDClient()
     {
         publish_all_ = true;
 
-        // TODO (Shawn): IS this really what we want to do?
         glog.is_warn() && glog << "No device configured. We will publish all GPS data."
                                << std::endl;
     }
 
-    gpsmm gps_rec(cfg().hostname().c_str(), std::to_string(cfg().port()).c_str());
-
-    if (!gps_rec.is_open())
+    if (!gps_rec_.is_open())
     {
         // Check constructor success.
         glog.is_die() && glog << "Could not connect to GPSD at " << cfg().hostname()
                               << " port: " << cfg().port() << std::endl;
     }
 
     // compare devices from our config to what gps_rec has in it's list.
-    if (gps_rec.stream(WATCH_ENABLE | WATCH_JSON) == NULL)
+    if (gps_rec_.stream(WATCH_ENABLE | WATCH_JSON) == NULL)
     {
-        glog.is_die() && glog << "No GPSD running.\n";
+        glog.is_die() && glog << "No GPSD running." << std::endl;
     }
+}
 
-    while (true)
-    {
-        // Notify data missing?
-        if (!gps_rec.waiting(1000))
-            continue;
+void goby::apps::zeromq::GPSDClient::loop()
+{
+    // Notify data missing?
+    if (!gps_rec_.waiting(1000))
```

Comment:
We might want to make this number bigger. 
it is how long to wait for data in microseconds. in the infinite loop, it wasn't too bad to have it so small, but assuming `loop()` is called at 1hz, do we want this to be 1 second also?